src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -48,25 +48,43 @@ def market_stats_callback(self, payload):
         self.publish_json(market_stats_response_channel, ret_val)
 
     def set_grid_fees_callback(self, payload):
-        market_stats_response_channel = f"{self.channel_prefix}/response/grid_fees"
+        grid_fees_response_channel = f"{self.channel_prefix}/response/grid_fees"
         payload_data = json.loads(payload["data"])
-        self.area.transfer_fee_const = payload_data["fee"]
-        self.publish_json(market_stats_response_channel, {
-            "status": "ready", "command": "grid_fees",
-            "market_fee_const": str(self.area.transfer_fee_const),
-            "transaction_id": payload_data.get("transaction_id", None)}
-         )
+        if "fee_const" in payload_data and payload_data["fee_const"] is not None and \
+                self.area.config.grid_fee_type == 1:
+            self.area.transfer_fee_const = payload_data["fee_const"]
+            self.publish_json(grid_fees_response_channel, {
+                "status": "ready", "command": "grid_fees",
+                "market_fee_const": str(self.area.transfer_fee_const),
+                "transaction_id": payload_data.get("transaction_id", None)}
+             )
+        elif "fee_percent" in payload_data and payload_data["fee_percent"] is not None and \
+                self.area.config.grid_fee_type == 2:
```

Comment:
I am missing some validation here, and in the const branch. The incoming fees should be in the range [0-100] for the percentage fees, [0-200] for the constant fees.

Reply:
Yes using `d3a-interface` grid_fee_validator